modules/imgcodecs/test/test_animation.cpp

```diff
@@ -676,7 +676,12 @@ TEST(Imgcodecs_APNG, imread_animation_16u)
     img = imread(filename, IMREAD_ANYDEPTH);
     ASSERT_FALSE(img.empty());
     EXPECT_TRUE(img.type() == CV_16UC1);
+#ifdef HAVE_PNG
     EXPECT_EQ(19519, img.at<ushort>(0, 0));
+#endif
+#ifdef HAVE_SPNG
+    EXPECT_EQ(19517, img.at<ushort>(0, 0));
+#endif
```

Comment:
Sounds like wrong rounding somewhere in libspng.

Reply:
last commit is temporarily to show that the spng conversion code is the same as libpng. i temporarily used `if (m_frame_count > 5)` that means the difference caused by cvtColor which i used in apng decoding.

```
[ RUN      ] Imgcodecs_APNG.imread_animation_16u
/build/precommit_linux64/4.x/opencv/modules/imgcodecs/test/test_animation.cpp:680: Failure
Expected equality of these values:
  19519
  img.at<ushort>(0, 0)
    Which is: 19517
[  FAILED  ] Imgcodecs_APNG.imread_animation_16u (2 ms)
```

Reply:
If you find it appropriate, I can add a to-do note and let's merge this. I'll work on apng and try to find a solution for this difference later.